### ISSUE TITLE
Additional caution for new cop: Style/SendWithLiteralMethodName

### ISSUE THREAD

**Issue by jasonkarns** [rubocop/rubocop#12983]
## Is your feature request related to a problem? Please describe.

I believe some additional caution should be urged for the new cop: `Style/SendWithLiteralMethodName` https://github.com/rubocop/rubocop/pull/12850

Presently the cop is marked unsafe for "receiver" reasons (which I don't completely understand).
However, there are other cases where simply converting from `send` (or `public_send`) to actual method invocation is not safe.

Writer methods in particular behave differently when invoked directly than with `send`.

```rb
class WeirdRuby
  attr_reader :wat
  def wat=(w)
    @wat = w
    42
  end
end
```

```irb
>> fun = WeirdRuby.new
=> #<WeirdRuby:0x000000011d9fa8d0>
>> fun.wat = 7
=> 7
>> fun.public_send(:wat=, 7)
=> 42
```

The exact same method, invoked via send vs invoked directly results in completely different return value. (IIRC it's because the ruby compiler special-cases `=` syntax to ensure the expression evaluates to the assigned value, even if the writer method returns something else. However, with `send` invocation, no such protection is in place.

## Describe the solution you'd like

I'm not really sure. Is it possible (or reasonable) for the cop to detect writer methods and refuse autocorrect them? Or should that be an additional option? Or should this just be an extra example in the docs? Or a test case in rubocop's test suite?

I would be happy to provide a PR depending on the approach desired. But I think _something_ should be done since this could be a huge footgun. (Not huge in frequency, but in debugging. No one would ever think to confirm that send vs direct-method could ever behave differently.)

## Additional context

This case may seem obscure (and it very likely is), but considering that if ever a writer method is indeed implemented directly instead of via the `attr_writer` macro, the odds that the return value differs from the "assigned" value go up greatly. (The only reason to override a writer is to do something odd anyway. And it's super common to have an "accidental" return value during a side-effect operation by not being careful of the last expression in the method. I've seen custom writers like this act as setters that are calling off to a service somehow and return nil or status codes or things completely unrelated to "assignment" value.)

**Comment by jasonkarns** [comment]
Possibly an improvement is to extend the method name regex from https://github.com/rubocop/rubocop/pull/12925 to exclude writers?
